fs2/model.py

```diff
@@ -256,6 +258,13 @@ def on_load_checkpoint(self, checkpoint):
         Note, this shouldn't fail on different versions of pydantic anymore,
         but it will fail on breaking changes to the config. We should catch those exceptions
         and handle them appropriately."""
+        if "model_info" in checkpoint:
+            assert (
+                checkpoint["model_info"]["name"] == FastSpeech2.__name__
+            ), f"""Wrong model type ({checkpoint["model_info"]["name"]}), we are expecting a { FastSpeech2.__name__ }"""
+            assert (
+                checkpoint["model_info"]["version"] == self.__version__
+            ), f"""Wrong model's version({checkpoint["model_info"]["version"]}), we are expecting version {FastSpeech2.__version__}"""
```

Comment:
Ici aussi ça prend qqc de plus friendly:
 - if version matches, we are happy
 - if version is lower than current and with a known value, apply whatever conversion we know to make it compatible with the current software. E.g., if 1.1 added field `foo` with default value `"bar"`, and the model is 1.0, initialize `foo="bar"` here.
 - if version is greater than current, I guess that's a `ValueError`, and it really ought to generate a message that says "Your model was created with a newer version of EveryVoice, please update your software." or something like that.

Reply:
And when `"model_info" not in checkpoint`, that should be treated as if version was v1, so that models people are creating now with EV, before this version info is being saved, should get all this logic working for them when we publish version v1.1 of the format.